source/lac/scalapack.cc

```diff
@@ -327,16 +327,29 @@ ScaLAPACKMatrix<NumberType>::copy_to(FullMatrix<NumberType> &matrix) const
   // we could move the following lines under the main loop above,
   // but they would be dependent on glob_i and glob_j, which
   // won't make it much prettier
-  if (property == LAPACKSupport::lower_triangular)
-    for (unsigned int i = 0; i < matrix.n(); ++i)
-      for (unsigned int j = i + 1; j < matrix.m(); ++j)
-        matrix(i, j) =
-          (state == LAPACKSupport::inverse_matrix ? matrix(j, i) : 0.);
-  else if (property == LAPACKSupport::upper_triangular)
-    for (unsigned int i = 0; i < matrix.n(); ++i)
-      for (unsigned int j = 0; j < i; ++j)
-        matrix(i, j) =
-          (state == LAPACKSupport::inverse_matrix ? matrix(j, i) : 0.);
+  if (state == LAPACKSupport::cholesky)
+    {
+      if (uplo == 'L')
```

Comment:
Please use `property == LAPACKSupport::lower_triangular` as on the exit from Cholesky it should be set. Same next loop.